{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XUZS4GUkHIEG"
      },
      "source": [
        "# Подготовка файлов и программ\n",
        "\n",
        "Некоторые программы ставятся дважды из разных источников для демонстрации различных вариантов в случае неполадок. Для оптимизации времени работы необходимо удалить лишний код."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "id": "lrtRbP3fAvj8"
      },
      "outputs": [],
      "source": [
        "pip install -q condacolab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "id": "jQ1Oqzb9BXoG"
      },
      "outputs": [],
      "source": [
        "import condacolab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "q5FUOHCgBZWp",
        "outputId": "071b6a75-91cf-42be-ea6d-9b657236d60c"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "✨🍰✨ Everything looks OK!\n"
          ]
        }
      ],
      "source": [
        "condacolab.install()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "id": "XnbIt77YBOT9",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "8e09ed69-654d-40c7-b514-9d319a471efa"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - seqtk\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    ca-certificates-2021.10.8  |       ha878542_0         139 KB  conda-forge\n",
            "    certifi-2021.10.8          |   py37h89c1867_1         145 KB  conda-forge\n",
            "    conda-4.11.0               |   py37h89c1867_1         983 KB  conda-forge\n",
            "    seqtk-1.3                  |       h5bf99c6_3          38 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:         1.3 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  seqtk              bioconda/linux-64::seqtk-1.3-h5bf99c6_3\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  ca-certificates                      2020.12.5-ha878542_0 --> 2021.10.8-ha878542_0\n",
            "  certifi                          2020.12.5-py37h89c1867_1 --> 2021.10.8-py37h89c1867_1\n",
            "  conda                                4.9.2-py37h89c1867_0 --> 4.11.0-py37h89c1867_1\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "seqtk-1.3            | 38 KB     | : 100% 1.0/1 [00:00<00:00,  7.10it/s]               \n",
            "ca-certificates-2021 | 139 KB    | : 100% 1.0/1 [00:00<00:00, 10.61it/s]\n",
            "conda-4.11.0         | 983 KB    | : 100% 1.0/1 [00:00<00:00,  2.88it/s]\n",
            "certifi-2021.10.8    | 145 KB    | : 100% 1.0/1 [00:00<00:00, 15.06it/s]\n",
            "Preparing transaction: / \b\bdone\n",
            "Verifying transaction: \\ \b\bdone\n",
            "Executing transaction: / \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda seqtk"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sCBV_31qHOik",
        "outputId": "dfc14d4b-fd1b-470a-a0f7-0001e320c40a"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - trimmomatic\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    openjdk-8.0.312            |       h7f98852_0        97.6 MB  conda-forge\n",
            "    python_abi-3.7             |          2_cp37m           4 KB  conda-forge\n",
            "    trimmomatic-0.39           |       hdfd78af_2         144 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        97.8 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  openjdk            conda-forge/linux-64::openjdk-8.0.312-h7f98852_0\n",
            "  trimmomatic        bioconda/noarch::trimmomatic-0.39-hdfd78af_2\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  python_abi                                    3.7-1_cp37m --> 3.7-2_cp37m\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "trimmomatic-0.39     | 144 KB    | : 100% 1.0/1 [00:00<00:00,  4.98it/s]                \n",
            "openjdk-8.0.312      | 97.6 MB   | : 100% 1.0/1 [00:15<00:00, 15.61s/it]\n",
            "python_abi-3.7       | 4 KB      | : 100% 1.0/1 [00:00<00:00, 21.07it/s]\n",
            "Preparing transaction: - \b\bdone\n",
            "Verifying transaction: | \b\bdone\n",
            "Executing transaction: - \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda trimmomatic"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "F0BLWsH2n3Ov",
        "outputId": "a242735f-2eb6-4699-8636-3bdd02c0095e"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting macs2\n",
            "  Downloading MACS2-2.2.7.1.tar.gz (2.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 2.1 MB 3.2 MB/s \n",
            "\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "    Preparing wheel metadata ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting intervene\n",
            "  Downloading intervene-0.6.5.tar.gz (5.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 5.5 MB 23.5 MB/s \n",
            "\u001b[?25hCollecting pybedtools\n",
            "  Downloading pybedtools-0.9.0.tar.gz (12.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 12.5 MB 31.3 MB/s \n",
            "\u001b[?25hCollecting matplotlib\n",
            "  Downloading matplotlib-3.5.1-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (11.2 MB)\n",
            "\u001b[K     |████████████████████████████████| 11.2 MB 51.7 MB/s \n",
            "\u001b[?25hCollecting pandas\n",
            "  Downloading pandas-1.3.5-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.3 MB)\n",
            "\u001b[K     |████████████████████████████████| 11.3 MB 52.5 MB/s \n",
            "\u001b[?25hCollecting numpy\n",
            "  Using cached numpy-1.21.5-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)\n",
            "Collecting scipy\n",
            "  Downloading scipy-1.7.3-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (38.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 38.1 MB 1.2 MB/s \n",
            "\u001b[?25hCollecting seaborn\n",
            "  Downloading seaborn-0.11.2-py3-none-any.whl (292 kB)\n",
            "\u001b[K     |████████████████████████████████| 292 kB 52.6 MB/s \n",
            "\u001b[?25hCollecting packaging>=20.0\n",
            "  Downloading packaging-21.3-py3-none-any.whl (40 kB)\n",
            "\u001b[K     |████████████████████████████████| 40 kB 4.8 MB/s \n",
            "\u001b[?25hCollecting pillow>=6.2.0\n",
            "  Downloading Pillow-9.0.1-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.3 MB)\n",
            "\u001b[K     |████████████████████████████████| 4.3 MB 45.1 MB/s \n",
            "\u001b[?25hCollecting python-dateutil>=2.7\n",
            "  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)\n",
            "\u001b[K     |████████████████████████████████| 247 kB 60.1 MB/s \n",
            "\u001b[?25hCollecting cycler>=0.10\n",
            "  Downloading cycler-0.11.0-py3-none-any.whl (6.4 kB)\n",
            "Collecting pyparsing>=2.2.1\n",
            "  Downloading pyparsing-3.0.7-py3-none-any.whl (98 kB)\n",
            "\u001b[K     |████████████████████████████████| 98 kB 6.2 MB/s \n",
            "\u001b[?25hCollecting kiwisolver>=1.0.1\n",
            "  Downloading kiwisolver-1.3.2-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.1 MB 43.9 MB/s \n",
            "\u001b[?25hCollecting fonttools>=4.22.0\n",
            "  Downloading fonttools-4.29.1-py3-none-any.whl (895 kB)\n",
            "\u001b[K     |████████████████████████████████| 895 kB 46.1 MB/s \n",
            "\u001b[?25hRequirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/site-packages (from python-dateutil>=2.7->matplotlib->intervene) (1.15.0)\n",
            "Collecting pytz>=2017.3\n",
            "  Downloading pytz-2021.3-py2.py3-none-any.whl (503 kB)\n",
            "\u001b[K     |████████████████████████████████| 503 kB 59.3 MB/s \n",
            "\u001b[?25hCollecting pysam\n",
            "  Downloading pysam-0.18.0-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (14.9 MB)\n",
            "\u001b[K     |████████████████████████████████| 14.9 MB 42.0 MB/s \n",
            "\u001b[?25hBuilding wheels for collected packages: intervene, macs2, pybedtools\n",
            "  Building wheel for intervene (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for intervene: filename=intervene-0.6.5-py3-none-any.whl size=5546118 sha256=7305e30f8cececb710afd22f43db75f997ca26170438d977f8c6f6f390c4abbf\n",
            "  Stored in directory: /root/.cache/pip/wheels/f3/d2/c1/18fa6122c36fdc731af4f16c59c356606b12a890a578294044\n",
            "  Building wheel for macs2 (PEP 517) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for macs2: filename=MACS2-2.2.7.1-cp37-cp37m-linux_x86_64.whl size=1790967 sha256=4a5714fa7e7701db74de68a7cb92645fe9815cbeedf95926086927d69239353b\n",
            "  Stored in directory: /root/.cache/pip/wheels/4e/34/10/f6222945f7c5b091e34b058a01a842ba9b1d960f2ab0c8e16f\n",
            "  Building wheel for pybedtools (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pybedtools: filename=pybedtools-0.9.0-cp37-cp37m-linux_x86_64.whl size=13654301 sha256=b5cf504a435ad69b3dcc217ececfc067cbcc1a2f4db2cf5ca3533408124aec35\n",
            "  Stored in directory: /root/.cache/pip/wheels/7a/44/0d/3a7449885adaf8ebb157da8c3c834a712f48b3b3b84ba51dda\n",
            "Successfully built intervene macs2 pybedtools\n",
            "Installing collected packages: pyparsing, pytz, python-dateutil, pillow, packaging, numpy, kiwisolver, fonttools, cycler, scipy, pysam, pandas, matplotlib, seaborn, pybedtools, macs2, intervene\n",
            "Successfully installed cycler-0.11.0 fonttools-4.29.1 intervene-0.6.5 kiwisolver-1.3.2 macs2-2.2.7.1 matplotlib-3.5.1 numpy-1.21.5 packaging-21.3 pandas-1.3.5 pillow-9.0.1 pybedtools-0.9.0 pyparsing-3.0.7 pysam-0.18.0 python-dateutil-2.8.2 pytz-2021.3 scipy-1.7.3 seaborn-0.11.2\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "cycler",
                  "dateutil",
                  "kiwisolver",
                  "numpy",
                  "pyparsing"
                ]
              }
            }
          },
          "metadata": {}
        }
      ],
      "source": [
        "!pip install macs2 intervene"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ce6hiSTdkarU",
        "outputId": "ac08d1a8-4442-4cc5-b1ed-5328006ac220"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following NEW packages will be installed:\n",
            "  bedtools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 39 not upgraded.\n",
            "Need to get 577 kB of archives.\n",
            "After this operation, 2,040 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bedtools amd64 2.26.0+dfsg-5 [577 kB]\n",
            "Fetched 577 kB in 1s (786 kB/s)\n",
            "Selecting previously unselected package bedtools.\n",
            "(Reading database ... 155332 files and directories currently installed.)\n",
            "Preparing to unpack .../bedtools_2.26.0+dfsg-5_amd64.deb ...\n",
            "Unpacking bedtools (2.26.0+dfsg-5) ...\n",
            "Setting up bedtools (2.26.0+dfsg-5) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bedtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AUML3dvsqT9E",
        "outputId": "8ebde459-6c84-4d66-d6a0-383bd32e9f89"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
            "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings\n",
            "Suggested packages:\n",
            "  bowtie2-examples python-cryptography-doc python-cryptography-vectors\n",
            "  python-enum34-doc python-genshi python-lxml python-lockfile-doc\n",
            "  python-openssl-doc python-openssl-dbg python-setuptools python-pyparsing-doc\n",
            "  python-rdflib-doc python-rdflib-tools python-socks python-ntlm\n",
            "The following NEW packages will be installed:\n",
            "  bowtie2 cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
            "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings samtools\n",
            "0 upgraded, 36 newly installed, 0 to remove and 39 not upgraded.\n",
            "Need to get 9,743 kB of archives.\n",
            "After this operation, 46.1 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pkg-resources all 39.0.1-2 [128 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-six all 1.11.0-2 [11.3 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-isodate all 0.6.0-1 [23.8 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pyparsing all 2.2.0+dfsg1-2 [52.1 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib all 4.2.1-2 [237 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-certifi all 2018.1.18-2 [144 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-chardet all 3.0.4-1 [80.3 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-idna all 2.6-1 [32.4 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-urllib3 all 1.22-1ubuntu0.18.04.2 [86.0 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-requests all 2.18.4-2ubuntu0.1 [58.5 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-ruamel.yaml amd64 0.15.34-1 [183 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-cachecontrol all 0.11.7-1 [15.2 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-lockfile all 1:0.12.2-2 [14.6 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-mistune all 0.8.3-2 [13.8 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib-jsonld all 0.4.0-3 [12.2 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-typing all 3.6.2-1 [22.2 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-avro all 1.8.2+dfsg-1 [23.2 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjs-bootstrap all 3.3.7+dfsg-2 [228 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-schema-salad all 2.6.20171201034858-3 [260 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-shellescape all 3.4.1-2 [4,432 B]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu bionic/universe amd64 cwltool all 1.0.20180302231433-1 [143 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libhts2 amd64 1.7-2 [300 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs-doc all 8.10.0~dfsg-2ubuntu0.4 [752 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-asn1crypto all 0.24.0-1 [72.7 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-cffi-backend amd64 1.11.5-1 [63.4 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-enum34 all 1.1.6-2 [34.8 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-ipaddress all 1.0.17-1 [18.2 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-cryptography amd64 2.1.4-1ubuntu1.4 [276 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-webencodings all 0.5-2 [10.3 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-html5lib all 0.999999999-1 [83.6 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-openssl all 17.5.0-1ubuntu1 [41.3 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-sparqlwrapper all 1.7.6-2 [21.8 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu bionic/universe amd64 samtools amd64 1.7-1 [288 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bowtie2 amd64 2.3.4.1-1 [1,174 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libc-ares2 amd64 1.14.0-1ubuntu0.1 [37.5 kB]\n",
            "Get:36 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs amd64 8.10.0~dfsg-2ubuntu0.4 [4,796 kB]\n",
            "Fetched 9,743 kB in 2s (6,185 kB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package python-pkg-resources.\n",
            "(Reading database ... 155383 files and directories currently installed.)\n",
            "Preparing to unpack .../00-python-pkg-resources_39.0.1-2_all.deb ...\n",
            "Unpacking python-pkg-resources (39.0.1-2) ...\n",
            "Selecting previously unselected package python-six.\n",
            "Preparing to unpack .../01-python-six_1.11.0-2_all.deb ...\n",
            "Unpacking python-six (1.11.0-2) ...\n",
            "Selecting previously unselected package python-isodate.\n",
            "Preparing to unpack .../02-python-isodate_0.6.0-1_all.deb ...\n",
            "Unpacking python-isodate (0.6.0-1) ...\n",
            "Selecting previously unselected package python-pyparsing.\n",
            "Preparing to unpack .../03-python-pyparsing_2.2.0+dfsg1-2_all.deb ...\n",
            "Unpacking python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Selecting previously unselected package python-rdflib.\n",
            "Preparing to unpack .../04-python-rdflib_4.2.1-2_all.deb ...\n",
            "Unpacking python-rdflib (4.2.1-2) ...\n",
            "Selecting previously unselected package python-certifi.\n",
            "Preparing to unpack .../05-python-certifi_2018.1.18-2_all.deb ...\n",
            "Unpacking python-certifi (2018.1.18-2) ...\n",
            "Selecting previously unselected package python-chardet.\n",
            "Preparing to unpack .../06-python-chardet_3.0.4-1_all.deb ...\n",
            "Unpacking python-chardet (3.0.4-1) ...\n",
            "Selecting previously unselected package python-idna.\n",
            "Preparing to unpack .../07-python-idna_2.6-1_all.deb ...\n",
            "Unpacking python-idna (2.6-1) ...\n",
            "Selecting previously unselected package python-urllib3.\n",
            "Preparing to unpack .../08-python-urllib3_1.22-1ubuntu0.18.04.2_all.deb ...\n",
            "Unpacking python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Selecting previously unselected package python-requests.\n",
            "Preparing to unpack .../09-python-requests_2.18.4-2ubuntu0.1_all.deb ...\n",
            "Unpacking python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Selecting previously unselected package python-ruamel.yaml.\n",
            "Preparing to unpack .../10-python-ruamel.yaml_0.15.34-1_amd64.deb ...\n",
            "Unpacking python-ruamel.yaml (0.15.34-1) ...\n",
            "Selecting previously unselected package python-cachecontrol.\n",
            "Preparing to unpack .../11-python-cachecontrol_0.11.7-1_all.deb ...\n",
            "Unpacking python-cachecontrol (0.11.7-1) ...\n",
            "Selecting previously unselected package python-lockfile.\n",
            "Preparing to unpack .../12-python-lockfile_1%3a0.12.2-2_all.deb ...\n",
            "Unpacking python-lockfile (1:0.12.2-2) ...\n",
            "Selecting previously unselected package python-mistune.\n",
            "Preparing to unpack .../13-python-mistune_0.8.3-2_all.deb ...\n",
            "Unpacking python-mistune (0.8.3-2) ...\n",
            "Selecting previously unselected package python-rdflib-jsonld.\n",
            "Preparing to unpack .../14-python-rdflib-jsonld_0.4.0-3_all.deb ...\n",
            "Unpacking python-rdflib-jsonld (0.4.0-3) ...\n",
            "Selecting previously unselected package python-typing.\n",
            "Preparing to unpack .../15-python-typing_3.6.2-1_all.deb ...\n",
            "Unpacking python-typing (3.6.2-1) ...\n",
            "Selecting previously unselected package python-avro.\n",
            "Preparing to unpack .../16-python-avro_1.8.2+dfsg-1_all.deb ...\n",
            "Unpacking python-avro (1.8.2+dfsg-1) ...\n",
            "Selecting previously unselected package libjs-bootstrap.\n",
            "Preparing to unpack .../17-libjs-bootstrap_3.3.7+dfsg-2_all.deb ...\n",
            "Unpacking libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Selecting previously unselected package python-schema-salad.\n",
            "Preparing to unpack .../18-python-schema-salad_2.6.20171201034858-3_all.deb ...\n",
            "Unpacking python-schema-salad (2.6.20171201034858-3) ...\n",
            "Selecting previously unselected package python-shellescape.\n",
            "Preparing to unpack .../19-python-shellescape_3.4.1-2_all.deb ...\n",
            "Unpacking python-shellescape (3.4.1-2) ...\n",
            "Selecting previously unselected package cwltool.\n",
            "Preparing to unpack .../20-cwltool_1.0.20180302231433-1_all.deb ...\n",
            "Unpacking cwltool (1.0.20180302231433-1) ...\n",
            "Selecting previously unselected package libhts2:amd64.\n",
            "Preparing to unpack .../21-libhts2_1.7-2_amd64.deb ...\n",
            "Unpacking libhts2:amd64 (1.7-2) ...\n",
            "Selecting previously unselected package nodejs-doc.\n",
            "Preparing to unpack .../22-nodejs-doc_8.10.0~dfsg-2ubuntu0.4_all.deb ...\n",
            "Unpacking nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Selecting previously unselected package python-asn1crypto.\n",
            "Preparing to unpack .../23-python-asn1crypto_0.24.0-1_all.deb ...\n",
            "Unpacking python-asn1crypto (0.24.0-1) ...\n",
            "Selecting previously unselected package python-cffi-backend.\n",
            "Preparing to unpack .../24-python-cffi-backend_1.11.5-1_amd64.deb ...\n",
            "Unpacking python-cffi-backend (1.11.5-1) ...\n",
            "Selecting previously unselected package python-enum34.\n",
            "Preparing to unpack .../25-python-enum34_1.1.6-2_all.deb ...\n",
            "Unpacking python-enum34 (1.1.6-2) ...\n",
            "Selecting previously unselected package python-ipaddress.\n",
            "Preparing to unpack .../26-python-ipaddress_1.0.17-1_all.deb ...\n",
            "Unpacking python-ipaddress (1.0.17-1) ...\n",
            "Selecting previously unselected package python-cryptography.\n",
            "Preparing to unpack .../27-python-cryptography_2.1.4-1ubuntu1.4_amd64.deb ...\n",
            "Unpacking python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Selecting previously unselected package python-webencodings.\n",
            "Preparing to unpack .../28-python-webencodings_0.5-2_all.deb ...\n",
            "Unpacking python-webencodings (0.5-2) ...\n",
            "Selecting previously unselected package python-html5lib.\n",
            "Preparing to unpack .../29-python-html5lib_0.999999999-1_all.deb ...\n",
            "Unpacking python-html5lib (0.999999999-1) ...\n",
            "Selecting previously unselected package python-openssl.\n",
            "Preparing to unpack .../30-python-openssl_17.5.0-1ubuntu1_all.deb ...\n",
            "Unpacking python-openssl (17.5.0-1ubuntu1) ...\n",
            "Selecting previously unselected package python-sparqlwrapper.\n",
            "Preparing to unpack .../31-python-sparqlwrapper_1.7.6-2_all.deb ...\n",
            "Unpacking python-sparqlwrapper (1.7.6-2) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../32-samtools_1.7-1_amd64.deb ...\n",
            "Unpacking samtools (1.7-1) ...\n",
            "Selecting previously unselected package bowtie2.\n",
            "Preparing to unpack .../33-bowtie2_2.3.4.1-1_amd64.deb ...\n",
            "Unpacking bowtie2 (2.3.4.1-1) ...\n",
            "Selecting previously unselected package libc-ares2:amd64.\n",
            "Preparing to unpack .../34-libc-ares2_1.14.0-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
            "Selecting previously unselected package nodejs.\n",
            "Preparing to unpack .../35-nodejs_8.10.0~dfsg-2ubuntu0.4_amd64.deb ...\n",
            "Unpacking nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-idna (2.6-1) ...\n",
            "Setting up nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-asn1crypto (0.24.0-1) ...\n",
            "Setting up python-mistune (0.8.3-2) ...\n",
            "Setting up python-certifi (2018.1.18-2) ...\n",
            "Setting up python-avro (1.8.2+dfsg-1) ...\n",
            "Setting up python-pkg-resources (39.0.1-2) ...\n",
            "Setting up bowtie2 (2.3.4.1-1) ...\n",
            "Setting up python-ruamel.yaml (0.15.34-1) ...\n",
            "Setting up python-cffi-backend (1.11.5-1) ...\n",
            "Setting up python-six (1.11.0-2) ...\n",
            "Setting up python-isodate (0.6.0-1) ...\n",
            "Setting up python-shellescape (3.4.1-2) ...\n",
            "Setting up python-enum34 (1.1.6-2) ...\n",
            "Setting up libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Setting up python-lockfile (1:0.12.2-2) ...\n",
            "Setting up python-ipaddress (1.0.17-1) ...\n",
            "Setting up libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
            "Setting up libhts2:amd64 (1.7-2) ...\n",
            "Setting up python-webencodings (0.5-2) ...\n",
            "Setting up python-typing (3.6.2-1) ...\n",
            "Setting up python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Setting up python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Setting up python-chardet (3.0.4-1) ...\n",
            "Setting up nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n",
            "Setting up python-rdflib (4.2.1-2) ...\n",
            "Setting up python-sparqlwrapper (1.7.6-2) ...\n",
            "Setting up python-rdflib-jsonld (0.4.0-3) ...\n",
            "Setting up python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Setting up python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Setting up python-cachecontrol (0.11.7-1) ...\n",
            "Setting up python-schema-salad (2.6.20171201034858-3) ...\n",
            "Setting up python-html5lib (0.999999999-1) ...\n",
            "Setting up samtools (1.7-1) ...\n",
            "Setting up python-openssl (17.5.0-1ubuntu1) ...\n",
            "Setting up cwltool (1.0.20180302231433-1) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1.3) ...\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libjawt.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libjli.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libnet.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/python3.7/dist-packages/ideep4py/lib/libmkldnn.so.0 is not a symbolic link\n",
            "\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bowtie2 samtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JgTq4lFjq93N",
        "outputId": "3c5e0f3e-d92e-4cb0-b487-8895f39c4482"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - bowtie2\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    bowtie2-2.3.5.1            |   py37he513fc3_0        12.0 MB  bioconda\n",
            "    perl-5.32.1                | 0_h7f98852_perl5        14.5 MB  conda-forge\n",
            "    tbb-2020.2                 |       h4bd325d_4         1.5 MB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        27.9 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  bowtie2            bioconda/linux-64::bowtie2-2.3.5.1-py37he513fc3_0\n",
            "  perl               conda-forge/linux-64::perl-5.32.1-0_h7f98852_perl5\n",
            "  tbb                conda-forge/linux-64::tbb-2020.2-h4bd325d_4\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "bowtie2-2.3.5.1      | 12.0 MB   | : 100% 1.0/1 [00:02<00:00,  2.30s/it]\n",
            "perl-5.32.1          | 14.5 MB   | : 100% 1.0/1 [00:04<00:00,  4.46s/it]               \n",
            "tbb-2020.2           | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  3.47it/s]\n",
            "Preparing transaction: \\ \b\bdone\n",
            "Verifying transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Executing transaction: / \b\b- \b\b\\ \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda bowtie2"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vcr5CjibHTgF"
      },
      "source": [
        "## Установка FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "62l1Q2ymbENA",
        "outputId": "cfa4ff1b-f5f6-4012-c6b4-ec4ac34769b0"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "openjdk version \"1.8.0_312\"\n",
            "OpenJDK Runtime Environment (Zulu 8.58.0.13-CA-linux64) (build 1.8.0_312-b07)\n",
            "OpenJDK 64-Bit Server VM (Zulu 8.58.0.13-CA-linux64) (build 25.312-b07, mixed mode)\n"
          ]
        }
      ],
      "source": [
        "!java -version"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8bpgwFgcbgRd",
        "outputId": "a6153270-c1ce-4450-81f3-25ac389f067f"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-10 06:46:38--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: ‘fastqc_v0.11.9.zip’\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  2.70MB/s    in 3.6s    \n",
            "\n",
            "2022-03-10 06:46:42 (2.70 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 14,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yTcTjOTJblcm",
        "outputId": "4ef3edec-c9f9-4a97-d13c-788da11ac142"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ],
      "source": [
        "!unzip fastqc_v0.11.9.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 15,
      "metadata": {
        "id": "S79LH0aXbvLx"
      },
      "outputs": [],
      "source": [
        "!chmod a+x FastQC/fastqc"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 16,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PsLuZDt4b7jT",
        "outputId": "131ce1c5-04e7-4480-8136-5125f95c9b9a"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "            FastQC - A high throughput sequence QC analysis tool\n",
            "\n",
            "SYNOPSIS\n",
            "\n",
            "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
            "\n",
            "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
            "           [-c contaminant file] seqfile1 .. seqfileN\n",
            "\n",
            "DESCRIPTION\n",
            "\n",
            "    FastQC reads a set of sequence files and produces from each one a quality\n",
            "    control report consisting of a number of different modules, each one of \n",
            "    which will help to identify a different potential type of problem in your\n",
            "    data.\n",
            "    \n",
            "    If no files to process are specified on the command line then the program\n",
            "    will start as an interactive graphical application.  If files are provided\n",
            "    on the command line then the program will run with no user interaction\n",
            "    required.  In this mode it is suitable for inclusion into a standardised\n",
            "    analysis pipeline.\n",
            "    \n",
            "    The options for the program as as follows:\n",
            "    \n",
            "    -h --help       Print this help file and exit\n",
            "    \n",
            "    -v --version    Print the version of the program and exit\n",
            "    \n",
            "    -o --outdir     Create all output files in the specified output directory.\n",
            "                    Please note that this directory must exist as the program\n",
            "                    will not create it.  If this option is not set then the \n",
            "                    output file for each sequence file is created in the same\n",
            "                    directory as the sequence file which was processed.\n",
            "                    \n",
            "    --casava        Files come from raw casava output. Files in the same sample\n",
            "                    group (differing only by the group number) will be analysed\n",
            "                    as a set rather than individually. Sequences with the filter\n",
            "                    flag set in the header will be excluded from the analysis.\n",
            "                    Files must have the same names given to them by casava\n",
            "                    (including being gzipped and ending with .gz) otherwise they\n",
            "                    won't be grouped together correctly.\n",
            "                    \n",
            "    --nano          Files come from nanopore sequences and are in fast5 format. In\n",
            "                    this mode you can pass in directories to process and the program\n",
            "                    will take in all fast5 files within those directories and produce\n",
            "                    a single output file from the sequences found in all files.                    \n",
            "                    \n",
            "    --nofilter      If running with --casava then don't remove read flagged by\n",
            "                    casava as poor quality when performing the QC analysis.\n",
            "                   \n",
            "    --extract       If set then the zipped output file will be uncompressed in\n",
            "                    the same directory after it has been created.  By default\n",
            "                    this option will be set if fastqc is run in non-interactive\n",
            "                    mode.\n",
            "                    \n",
            "    -j --java       Provides the full path to the java binary you want to use to\n",
            "                    launch fastqc. If not supplied then java is assumed to be in\n",
            "                    your path.\n",
            "                   \n",
            "    --noextract     Do not uncompress the output file after creating it.  You\n",
            "                    should set this option if you do not wish to uncompress\n",
            "                    the output when running in non-interactive mode.\n",
            "                    \n",
            "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
            "                    show data for every base in the read.  WARNING: Using this\n",
            "                    option will cause fastqc to crash and burn if you use it on\n",
            "                    really long reads, and your plots may end up a ridiculous size.\n",
            "                    You have been warned!\n",
            "                    \n",
            "    --min_length    Sets an artificial lower limit on the length of the sequence\n",
            "                    to be shown in the report.  As long as you set this to a value\n",
            "                    greater or equal to your longest read length then this will be\n",
            "                    the sequence length used to create your read groups.  This can\n",
            "                    be useful for making directly comaparable statistics from \n",
            "                    datasets with somewhat variable read lengths.\n",
            "                    \n",
            "    -f --format     Bypasses the normal sequence file format detection and\n",
            "                    forces the program to use the specified format.  Valid\n",
            "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
            "                    \n",
            "    -t --threads    Specifies the number of files which can be processed\n",
            "                    simultaneously.  Each thread will be allocated 250MB of\n",
            "                    memory so you shouldn't run more threads than your\n",
            "                    available memory will cope with, and not more than\n",
            "                    6 threads on a 32 bit machine\n",
            "                  \n",
            "    -c              Specifies a non-default file which contains the list of\n",
            "    --contaminants  contaminants to screen overrepresented sequences against.\n",
            "                    The file must contain sets of named contaminants in the\n",
            "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
            "                    be ignored.\n",
            "\n",
            "    -a              Specifies a non-default file which contains the list of\n",
            "    --adapters      adapter sequences which will be explicity searched against\n",
            "                    the library. The file must contain sets of named adapters\n",
            "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
            "                    will be ignored.\n",
            "                    \n",
            "    -l              Specifies a non-default file which contains a set of criteria\n",
            "    --limits        which will be used to determine the warn/error limits for the\n",
            "                    various modules.  This file can also be used to selectively \n",
            "                    remove some modules from the output all together.  The format\n",
            "                    needs to mirror the default limits.txt file found in the\n",
            "                    Configuration folder.\n",
            "                    \n",
            "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
            "                    module. Specified Kmer length must be between 2 and 10. Default\n",
            "                    length is 7 if not specified.\n",
            "                    \n",
            "   -q --quiet       Supress all progress messages on stdout and only report errors.\n",
            "   \n",
            "   -d --dir         Selects a directory to be used for temporary files written when\n",
            "                    generating report images. Defaults to system temp directory if\n",
            "                    not specified.\n",
            "                    \n",
            "BUGS\n",
            "\n",
            "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
            "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
            "                   \n",
            "    \n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc --help"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "10Fs3qrmHmJi"
      },
      "source": [
        "# Выравнивание"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-z4GQiMfU1wf"
      },
      "source": [
        "## Скачиваем чтения\n",
        "\n",
        "Для примера чтения были взяты из эксперимента https://www.encodeproject.org/experiments/ENCSR000AKF/"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lFWnMg0uU0sJ",
        "outputId": "3de79952-818b-4804-f847-14e93956448e"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-01 19:52:03--  https://www.encodeproject.org/files/ENCFF000ASM/@@download/ENCFF000ASM.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2010/11/16/9e6a6ac5-1dd3-4809-850e-1b0734d54a05/ENCFF000ASM.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF000ASM.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXX4VPAH6H&Signature=CdCTDg1f9AoL%2B0rOGe%2BtYXvBW1I%3D&x-amz-security-token=IQoJb3JpZ2luX2VjELT%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQCpVh%2B7Yb5ElaLWoDpKsoUXbAmT%2BbPqh89S7oEIrjaYjgIgc3ekyuPQkUKBKo0ub4tI6BvC6oLPQ%2BCQsLwVYIcY4P8q%2BgMIHBAAGgwyMjA3NDg3MTQ4NjMiDM7mlGowGi3H%2FFfj3SrXA4rtUMFNS6qI8jBJanpHwUUSFgBbajnd7houZ56PIKJTIdCd6%2BD7px%2FKGcabyQosbv6HxdYzO9RWI2Nqbe4lwo8dsS8SvS6s1Mle97I1p99RZz2NY9J5tmpGzKv4VdbqeHFLPkYchZgAOPcsp9ZcpDO%2FCEd9tqtPJt2p7mgi3VNjRLZCnHehG95OZbBGTui7FqGy%2Bl6A17RIEVj%2B3GJIX6evCQdV2iPjvFI64n3%2FTNcPyNaDTCIgBCJ4vspW85GMrohGzYuOBqy904Du3h1eoXUHb%2BJ5eHlTEdef0rwfsx0dUcyl0ErqmT7LrFaBM6C4BSYpHLW5mKCo2C9%2FymgrA%2Bh1dftCmSpD5IgMzVW3gjNjBkbqu2zWUk3Mt%2F8I5jPDsUjZwpkIWbndMdTq2NaQuOgqTio1YM0olaawpZd89uIOVj%2BW27GMz%2ByYx0K4GbhZZen6juyKt8rrrWD1S%2FfRksVqqej7z0EhOdU4YMrxsjQ3oU%2FLMELFYlwjs1IVScIBNVbmVlEG5oGM8cjWM%2BFJvUExmHIbA1FOd%2FCh4ISqcaAE%2FyOs861A730geOHhUdCT05H8DSEbeMN3Fq4LfkxL%2FGl6aG5F6%2FXJg7rSGgEgONdfshoaLhj7wDDH5PmQBjqlARiqkAzT0dq%2B1YNNmsdxyfpj83Zjw%2FfeQ6ZU6vYXnfjl6XT4AJVIV3qVmk7lV27AAcHxH21S2XAp4d5Y0QD6qFEN5Br5wKP4sHCJ%2FL8bXj9HlMOJFmaPMQEvZTwWFSgJy5C4ipPAbcVSdUvxP13FnCOWJdoupeFlXh%2FIjtl%2BjoWsfGPtwbMWnJFpI%2BsnWuyIs5uXluzzjDPb%2Fah9AqdckUxl7spJCw%3D%3D&Expires=1646293924 [following]\n",
            "--2022-03-01 19:52:04--  https://encode-public.s3.amazonaws.com/2010/11/16/9e6a6ac5-1dd3-4809-850e-1b0734d54a05/ENCFF000ASM.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF000ASM.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXX4VPAH6H&Signature=CdCTDg1f9AoL%2B0rOGe%2BtYXvBW1I%3D&x-amz-security-token=IQoJb3JpZ2luX2VjELT%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQCpVh%2B7Yb5ElaLWoDpKsoUXbAmT%2BbPqh89S7oEIrjaYjgIgc3ekyuPQkUKBKo0ub4tI6BvC6oLPQ%2BCQsLwVYIcY4P8q%2BgMIHBAAGgwyMjA3NDg3MTQ4NjMiDM7mlGowGi3H%2FFfj3SrXA4rtUMFNS6qI8jBJanpHwUUSFgBbajnd7houZ56PIKJTIdCd6%2BD7px%2FKGcabyQosbv6HxdYzO9RWI2Nqbe4lwo8dsS8SvS6s1Mle97I1p99RZz2NY9J5tmpGzKv4VdbqeHFLPkYchZgAOPcsp9ZcpDO%2FCEd9tqtPJt2p7mgi3VNjRLZCnHehG95OZbBGTui7FqGy%2Bl6A17RIEVj%2B3GJIX6evCQdV2iPjvFI64n3%2FTNcPyNaDTCIgBCJ4vspW85GMrohGzYuOBqy904Du3h1eoXUHb%2BJ5eHlTEdef0rwfsx0dUcyl0ErqmT7LrFaBM6C4BSYpHLW5mKCo2C9%2FymgrA%2Bh1dftCmSpD5IgMzVW3gjNjBkbqu2zWUk3Mt%2F8I5jPDsUjZwpkIWbndMdTq2NaQuOgqTio1YM0olaawpZd89uIOVj%2BW27GMz%2ByYx0K4GbhZZen6juyKt8rrrWD1S%2FfRksVqqej7z0EhOdU4YMrxsjQ3oU%2FLMELFYlwjs1IVScIBNVbmVlEG5oGM8cjWM%2BFJvUExmHIbA1FOd%2FCh4ISqcaAE%2FyOs861A730geOHhUdCT05H8DSEbeMN3Fq4LfkxL%2FGl6aG5F6%2FXJg7rSGgEgONdfshoaLhj7wDDH5PmQBjqlARiqkAzT0dq%2B1YNNmsdxyfpj83Zjw%2FfeQ6ZU6vYXnfjl6XT4AJVIV3qVmk7lV27AAcHxH21S2XAp4d5Y0QD6qFEN5Br5wKP4sHCJ%2FL8bXj9HlMOJFmaPMQEvZTwWFSgJy5C4ipPAbcVSdUvxP13FnCOWJdoupeFlXh%2FIjtl%2BjoWsfGPtwbMWnJFpI%2BsnWuyIs5uXluzzjDPb%2Fah9AqdckUxl7spJCw%3D%3D&Expires=1646293924\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.132.41\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.132.41|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1631313895 (1.5G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF000ASM.fastq.gz’\n",
            "\n",
            "ENCFF000ASM.fastq.g 100%[===================>]   1.52G  33.1MB/s    in 49s     \n",
            "\n",
            "2022-03-01 19:52:54 (31.6 MB/s) - ‘ENCFF000ASM.fastq.gz’ saved [1631313895/1631313895]\n",
            "\n",
            "--2022-03-01 19:52:54--  https://www.encodeproject.org/files/ENCFF000ATK/@@download/ENCFF000ATK.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2010/11/16/3ef85292-90e1-4694-a0d3-362cde3876a2/ENCFF000ATK.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF000ATK.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXX4VPAH6H&Signature=kDmkcvs4NVBRDtFH%2FgC8BURWW%2BU%3D&x-amz-security-token=IQoJb3JpZ2luX2VjELT%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQCpVh%2B7Yb5ElaLWoDpKsoUXbAmT%2BbPqh89S7oEIrjaYjgIgc3ekyuPQkUKBKo0ub4tI6BvC6oLPQ%2BCQsLwVYIcY4P8q%2BgMIHBAAGgwyMjA3NDg3MTQ4NjMiDM7mlGowGi3H%2FFfj3SrXA4rtUMFNS6qI8jBJanpHwUUSFgBbajnd7houZ56PIKJTIdCd6%2BD7px%2FKGcabyQosbv6HxdYzO9RWI2Nqbe4lwo8dsS8SvS6s1Mle97I1p99RZz2NY9J5tmpGzKv4VdbqeHFLPkYchZgAOPcsp9ZcpDO%2FCEd9tqtPJt2p7mgi3VNjRLZCnHehG95OZbBGTui7FqGy%2Bl6A17RIEVj%2B3GJIX6evCQdV2iPjvFI64n3%2FTNcPyNaDTCIgBCJ4vspW85GMrohGzYuOBqy904Du3h1eoXUHb%2BJ5eHlTEdef0rwfsx0dUcyl0ErqmT7LrFaBM6C4BSYpHLW5mKCo2C9%2FymgrA%2Bh1dftCmSpD5IgMzVW3gjNjBkbqu2zWUk3Mt%2F8I5jPDsUjZwpkIWbndMdTq2NaQuOgqTio1YM0olaawpZd89uIOVj%2BW27GMz%2ByYx0K4GbhZZen6juyKt8rrrWD1S%2FfRksVqqej7z0EhOdU4YMrxsjQ3oU%2FLMELFYlwjs1IVScIBNVbmVlEG5oGM8cjWM%2BFJvUExmHIbA1FOd%2FCh4ISqcaAE%2FyOs861A730geOHhUdCT05H8DSEbeMN3Fq4LfkxL%2FGl6aG5F6%2FXJg7rSGgEgONdfshoaLhj7wDDH5PmQBjqlARiqkAzT0dq%2B1YNNmsdxyfpj83Zjw%2FfeQ6ZU6vYXnfjl6XT4AJVIV3qVmk7lV27AAcHxH21S2XAp4d5Y0QD6qFEN5Br5wKP4sHCJ%2FL8bXj9HlMOJFmaPMQEvZTwWFSgJy5C4ipPAbcVSdUvxP13FnCOWJdoupeFlXh%2FIjtl%2BjoWsfGPtwbMWnJFpI%2BsnWuyIs5uXluzzjDPb%2Fah9AqdckUxl7spJCw%3D%3D&Expires=1646293974 [following]\n",
            "--2022-03-01 19:52:54--  https://encode-public.s3.amazonaws.com/2010/11/16/3ef85292-90e1-4694-a0d3-362cde3876a2/ENCFF000ATK.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF000ATK.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXX4VPAH6H&Signature=kDmkcvs4NVBRDtFH%2FgC8BURWW%2BU%3D&x-amz-security-token=IQoJb3JpZ2luX2VjELT%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQCpVh%2B7Yb5ElaLWoDpKsoUXbAmT%2BbPqh89S7oEIrjaYjgIgc3ekyuPQkUKBKo0ub4tI6BvC6oLPQ%2BCQsLwVYIcY4P8q%2BgMIHBAAGgwyMjA3NDg3MTQ4NjMiDM7mlGowGi3H%2FFfj3SrXA4rtUMFNS6qI8jBJanpHwUUSFgBbajnd7houZ56PIKJTIdCd6%2BD7px%2FKGcabyQosbv6HxdYzO9RWI2Nqbe4lwo8dsS8SvS6s1Mle97I1p99RZz2NY9J5tmpGzKv4VdbqeHFLPkYchZgAOPcsp9ZcpDO%2FCEd9tqtPJt2p7mgi3VNjRLZCnHehG95OZbBGTui7FqGy%2Bl6A17RIEVj%2B3GJIX6evCQdV2iPjvFI64n3%2FTNcPyNaDTCIgBCJ4vspW85GMrohGzYuOBqy904Du3h1eoXUHb%2BJ5eHlTEdef0rwfsx0dUcyl0ErqmT7LrFaBM6C4BSYpHLW5mKCo2C9%2FymgrA%2Bh1dftCmSpD5IgMzVW3gjNjBkbqu2zWUk3Mt%2F8I5jPDsUjZwpkIWbndMdTq2NaQuOgqTio1YM0olaawpZd89uIOVj%2BW27GMz%2ByYx0K4GbhZZen6juyKt8rrrWD1S%2FfRksVqqej7z0EhOdU4YMrxsjQ3oU%2FLMELFYlwjs1IVScIBNVbmVlEG5oGM8cjWM%2BFJvUExmHIbA1FOd%2FCh4ISqcaAE%2FyOs861A730geOHhUdCT05H8DSEbeMN3Fq4LfkxL%2FGl6aG5F6%2FXJg7rSGgEgONdfshoaLhj7wDDH5PmQBjqlARiqkAzT0dq%2B1YNNmsdxyfpj83Zjw%2FfeQ6ZU6vYXnfjl6XT4AJVIV3qVmk7lV27AAcHxH21S2XAp4d5Y0QD6qFEN5Br5wKP4sHCJ%2FL8bXj9HlMOJFmaPMQEvZTwWFSgJy5C4ipPAbcVSdUvxP13FnCOWJdoupeFlXh%2FIjtl%2BjoWsfGPtwbMWnJFpI%2BsnWuyIs5uXluzzjDPb%2Fah9AqdckUxl7spJCw%3D%3D&Expires=1646293974\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.253.91\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.253.91|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 762051363 (727M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF000ATK.fastq.gz’\n",
            "\n",
            "ENCFF000ATK.fastq.g 100%[===================>] 726.75M  35.0MB/s    in 22s     \n",
            "\n",
            "2022-03-01 19:53:16 (33.4 MB/s) - ‘ENCFF000ATK.fastq.gz’ saved [762051363/762051363]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "# ChIP-seq на гистоновой метке\n",
        "!wget https://www.encodeproject.org/files/ENCFF000ASM/@@download/ENCFF000ASM.fastq.gz\n",
        "!wget https://www.encodeproject.org/files/ENCFF000ATK/@@download/ENCFF000ATK.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# ChIP-seq контроль\n",
        "!wget https://www.encodeproject.org/files/ENCFF000ARK/@@download/ENCFF000ARK.fastq.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "gR3EcgcgBgY5",
        "outputId": "aa3918d0-ae31-423d-da67-de3f19b4e71f"
      },
      "execution_count": 17,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-10 06:47:05--  https://www.encodeproject.org/files/ENCFF000ARK/@@download/ENCFF000ARK.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2010/11/16/55a84578-56e7-4ea7-a60a-7a150e21201c/ENCFF000ARK.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF000ARK.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXSWXXRSXN&Signature=GztOQtN9z4SsFpJyER2nthYZhlw%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEH8aCXVzLXdlc3QtMiJHMEUCIFHtTaCUqSYv5hUxZm9gBAyNVRd5tBC0pXOArnIiuMgjAiEAon5J%2BiF1kuySaiX5aflju%2BbxgvU4wP9O9Q8aAS1MNqMqgwQI6P%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDLC8bI3F9NQLfmPbHirXAwObXb%2BXBI0MFBcjG47WteIPLGkUcjwGPXWhpvZZnLB%2FlVCC%2Fbkbj8mUDBbiL8v%2Fdl6ZJeXMk3SfR9%2FbfFlQx0iFtHoOU5rNNI%2BgO8irCg4ebNgzvC7bQ28PlbZVoMsLXX9KEv6%2BhKm7zlwQsgqIUPhS1P%2BNJ3c9mlHX3GI66B5y73398aWpjz3m15UOEJt9cRhQBx%2BRSlqbZ2OnpH4USPFjb0x2NmNq%2BnOJMEcUYKariFrorUWKszL5UQnk42TSWpafDIe5%2F7cdwVFypqZ%2BZSNLU1rOPfd2hEKst%2Btzji8a1I5IpPGGHAmP9h08yR5kU%2FONg6XyjvV6BfPlgenwRPhP3TiylzjCS%2FhJO%2Fn%2FlRyPdurM4V3CGt2Vlxyh1jE%2BSnei9bt6E4gMlx5Y%2FF%2B2dE0iru7TVxY9mItD8xhXj5E5mHyUnp95vHRUPrRFoRfP52uxcWumYqknHD%2BcY5HZbBdzlN6TAEAHuouAS2%2Fn0fEFWl88rYToYmM64t4v3RQ%2BzNfOEtMnBoJpPkvG0%2BNWlIz77DCDnHJv1148fWMQDITiD4IHLkFt%2BNvNXysccMmuLsuOLJ3TKmpxQryPUAkZYpmOJz95jkH0d0qu1Bof9eJBbsK3XHbywTCSsqaRBjqlATX%2FzSb%2BTPmfE27DyNAxpbAkZ2u5keYtkcctR8I9TWybUn7pszhBu3wTPO0q8zjkZQIGDFkUtL2gGdqNj0bDEraCZiQu6cuoveH1hyoU%2BC8VrmuB8MHi%2Bj2e75W2aHMb4rqjbLvifK43BNzYCLsVMtbaOoR6QASlw8IvoK9mbW00cWhMa07ty8JJkzci%2FCfDumf45AN3DXfQwfghz7mR23Aw6yAygQ%3D%3D&Expires=1647024425 [following]\n",
            "--2022-03-10 06:47:05--  https://encode-public.s3.amazonaws.com/2010/11/16/55a84578-56e7-4ea7-a60a-7a150e21201c/ENCFF000ARK.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF000ARK.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXSWXXRSXN&Signature=GztOQtN9z4SsFpJyER2nthYZhlw%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEH8aCXVzLXdlc3QtMiJHMEUCIFHtTaCUqSYv5hUxZm9gBAyNVRd5tBC0pXOArnIiuMgjAiEAon5J%2BiF1kuySaiX5aflju%2BbxgvU4wP9O9Q8aAS1MNqMqgwQI6P%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDLC8bI3F9NQLfmPbHirXAwObXb%2BXBI0MFBcjG47WteIPLGkUcjwGPXWhpvZZnLB%2FlVCC%2Fbkbj8mUDBbiL8v%2Fdl6ZJeXMk3SfR9%2FbfFlQx0iFtHoOU5rNNI%2BgO8irCg4ebNgzvC7bQ28PlbZVoMsLXX9KEv6%2BhKm7zlwQsgqIUPhS1P%2BNJ3c9mlHX3GI66B5y73398aWpjz3m15UOEJt9cRhQBx%2BRSlqbZ2OnpH4USPFjb0x2NmNq%2BnOJMEcUYKariFrorUWKszL5UQnk42TSWpafDIe5%2F7cdwVFypqZ%2BZSNLU1rOPfd2hEKst%2Btzji8a1I5IpPGGHAmP9h08yR5kU%2FONg6XyjvV6BfPlgenwRPhP3TiylzjCS%2FhJO%2Fn%2FlRyPdurM4V3CGt2Vlxyh1jE%2BSnei9bt6E4gMlx5Y%2FF%2B2dE0iru7TVxY9mItD8xhXj5E5mHyUnp95vHRUPrRFoRfP52uxcWumYqknHD%2BcY5HZbBdzlN6TAEAHuouAS2%2Fn0fEFWl88rYToYmM64t4v3RQ%2BzNfOEtMnBoJpPkvG0%2BNWlIz77DCDnHJv1148fWMQDITiD4IHLkFt%2BNvNXysccMmuLsuOLJ3TKmpxQryPUAkZYpmOJz95jkH0d0qu1Bof9eJBbsK3XHbywTCSsqaRBjqlATX%2FzSb%2BTPmfE27DyNAxpbAkZ2u5keYtkcctR8I9TWybUn7pszhBu3wTPO0q8zjkZQIGDFkUtL2gGdqNj0bDEraCZiQu6cuoveH1hyoU%2BC8VrmuB8MHi%2Bj2e75W2aHMb4rqjbLvifK43BNzYCLsVMtbaOoR6QASlw8IvoK9mbW00cWhMa07ty8JJkzci%2FCfDumf45AN3DXfQwfghz7mR23Aw6yAygQ%3D%3D&Expires=1647024425\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.170.57\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.170.57|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 502826550 (480M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF000ARK.fastq.gz’\n",
            "\n",
            "ENCFF000ARK.fastq.g 100%[===================>] 479.53M  32.9MB/s    in 16s     \n",
            "\n",
            "2022-03-10 06:47:21 (30.9 MB/s) - ‘ENCFF000ARK.fastq.gz’ saved [502826550/502826550]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "v73fH7WiVDu9"
      },
      "outputs": [],
      "source": [
        "# распаковка ахрива\n",
        "!gzip -d /content/ENCFF000ASM.fastq.gz /content/ENCFF000ATK.fastq.gz /content/ENCFF000ARK.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "6sPcIefNHx-L"
      },
      "source": [
        "## FastQC (или multiQC может быть использован в работе)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5XM6jBcuGjMV",
        "outputId": "0b2f78ab-40b5-4f35-b2ff-2cf78a921f43"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF000ASM.fastq\n",
            "Approx 5% complete for ENCFF000ASM.fastq\n",
            "Approx 10% complete for ENCFF000ASM.fastq\n",
            "Approx 15% complete for ENCFF000ASM.fastq\n",
            "Approx 20% complete for ENCFF000ASM.fastq\n",
            "Approx 25% complete for ENCFF000ASM.fastq\n",
            "Approx 30% complete for ENCFF000ASM.fastq\n",
            "Approx 35% complete for ENCFF000ASM.fastq\n",
            "Approx 40% complete for ENCFF000ASM.fastq\n",
            "Approx 45% complete for ENCFF000ASM.fastq\n",
            "Approx 50% complete for ENCFF000ASM.fastq\n",
            "Approx 55% complete for ENCFF000ASM.fastq\n",
            "Approx 60% complete for ENCFF000ASM.fastq\n",
            "Approx 65% complete for ENCFF000ASM.fastq\n",
            "Approx 70% complete for ENCFF000ASM.fastq\n",
            "Approx 75% complete for ENCFF000ASM.fastq\n",
            "Approx 80% complete for ENCFF000ASM.fastq\n",
            "Approx 85% complete for ENCFF000ASM.fastq\n",
            "Approx 90% complete for ENCFF000ASM.fastq\n",
            "Approx 95% complete for ENCFF000ASM.fastq\n",
            "Analysis complete for ENCFF000ASM.fastq\n",
            "Started analysis of ENCFF000ATK.fastq\n",
            "Approx 5% complete for ENCFF000ATK.fastq\n",
            "Approx 10% complete for ENCFF000ATK.fastq\n",
            "Approx 15% complete for ENCFF000ATK.fastq\n",
            "Approx 20% complete for ENCFF000ATK.fastq\n",
            "Approx 25% complete for ENCFF000ATK.fastq\n",
            "Approx 30% complete for ENCFF000ATK.fastq\n",
            "Approx 35% complete for ENCFF000ATK.fastq\n",
            "Approx 40% complete for ENCFF000ATK.fastq\n",
            "Approx 45% complete for ENCFF000ATK.fastq\n",
            "Approx 50% complete for ENCFF000ATK.fastq\n",
            "Approx 55% complete for ENCFF000ATK.fastq\n",
            "Approx 60% complete for ENCFF000ATK.fastq\n",
            "Approx 65% complete for ENCFF000ATK.fastq\n",
            "Approx 70% complete for ENCFF000ATK.fastq\n",
            "Approx 75% complete for ENCFF000ATK.fastq\n",
            "Approx 80% complete for ENCFF000ATK.fastq\n",
            "Approx 85% complete for ENCFF000ATK.fastq\n",
            "Approx 90% complete for ENCFF000ATK.fastq\n",
            "Approx 95% complete for ENCFF000ATK.fastq\n",
            "Analysis complete for ENCFF000ATK.fastq\n",
            "Started analysis of ENCFF000ARK.fastq\n",
            "Approx 5% complete for ENCFF000ARK.fastq\n",
            "Approx 10% complete for ENCFF000ARK.fastq\n",
            "Approx 15% complete for ENCFF000ARK.fastq\n",
            "Approx 20% complete for ENCFF000ARK.fastq\n",
            "Approx 25% complete for ENCFF000ARK.fastq\n",
            "Approx 30% complete for ENCFF000ARK.fastq\n",
            "Approx 35% complete for ENCFF000ARK.fastq\n",
            "Approx 40% complete for ENCFF000ARK.fastq\n",
            "Approx 45% complete for ENCFF000ARK.fastq\n",
            "Approx 50% complete for ENCFF000ARK.fastq\n",
            "Approx 55% complete for ENCFF000ARK.fastq\n",
            "Approx 60% complete for ENCFF000ARK.fastq\n",
            "Approx 65% complete for ENCFF000ARK.fastq\n",
            "Approx 70% complete for ENCFF000ARK.fastq\n",
            "Approx 75% complete for ENCFF000ARK.fastq\n",
            "Approx 80% complete for ENCFF000ARK.fastq\n",
            "Approx 85% complete for ENCFF000ARK.fastq\n",
            "Approx 90% complete for ENCFF000ARK.fastq\n",
            "Approx 95% complete for ENCFF000ARK.fastq\n",
            "Analysis complete for ENCFF000ARK.fastq\n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc /content/ENCFF000ASM.fastq\n",
        "!./FastQC/fastqc /content/ENCFF000ATK.fastq\n",
        "!./FastQC/fastqc /content/ENCFF000ARK.fastq"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-TO3DQaUoGhB"
      },
      "source": [
        "Далее необходимо проанализировать выдачу fastqc, произвести фильтрацию или подрезание чтений в случае необходимости. Для примера указано использование trimmomatic, также можно использовать seqtk."
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!trimmomatic SE -phred33 ENCFF000ARK.fastq ENCFF000ARK_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
        "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "wq2xnkd9lq7m",
        "outputId": "d78f098b-ba15-4c7e-902d-1f6e3a2c2e19"
      },
      "execution_count": 20,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TrimmomaticSE: Started with arguments:\n",
            " -phred33 ENCFF000ARK.fastq ENCFF000ARK_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Automatically using 2 threads\n",
            "java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)\n",
            "\tat java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.io.FileInputStream.open(FileInputStream.java:195)\n",
            "\tat java.io.FileInputStream.<init>(FileInputStream.java:138)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
            "Input Reads: 10676160 Surviving: 6446684 (60.38%) Dropped: 4229476 (39.62%)\n",
            "TrimmomaticSE: Completed successfully\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!./FastQC/fastqc ENCFF000ARK_trimmed.fastq"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "B_a1IdNalrHz",
        "outputId": "0c54c37f-eb3a-4275-ec7c-0e3609dbc962"
      },
      "execution_count": 21,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF000ARK_trimmed.fastq\n",
            "Approx 5% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 10% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 15% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 20% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 25% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 30% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 35% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 40% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 45% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 50% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 55% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 60% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 65% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 70% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 75% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 80% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 85% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 90% complete for ENCFF000ARK_trimmed.fastq\n",
            "Approx 95% complete for ENCFF000ARK_trimmed.fastq\n",
            "Analysis complete for ENCFF000ARK_trimmed.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Qellza2PI7fT",
        "outputId": "117c8f3c-f18a-4c07-8de8-7e807f5429c1"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TrimmomaticSE: Started with arguments:\n",
            " -phred33 ENCFF000ATK.fastq ENCFF000ATK_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Automatically using 2 threads\n",
            "java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)\n",
            "\tat java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.io.FileInputStream.open(FileInputStream.java:195)\n",
            "\tat java.io.FileInputStream.<init>(FileInputStream.java:138)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
            "Input Reads: 14936579 Surviving: 8383067 (56.12%) Dropped: 6553512 (43.88%)\n",
            "TrimmomaticSE: Completed successfully\n"
          ]
        }
      ],
      "source": [
        "!trimmomatic SE -phred33 ENCFF000ATK.fastq ENCFF000ATK_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
        "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PWBBF50gI7mI",
        "outputId": "7966289c-db69-4454-bf2c-70a3081dca62"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF000ATK_trimmed.fastq\n",
            "Approx 5% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 10% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 15% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 20% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 25% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 30% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 35% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 40% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 45% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 50% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 55% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 60% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 65% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 70% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 75% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 80% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 85% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 90% complete for ENCFF000ATK_trimmed.fastq\n",
            "Approx 95% complete for ENCFF000ATK_trimmed.fastq\n",
            "Analysis complete for ENCFF000ATK_trimmed.fastq\n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc ENCFF000ATK_trimmed.fastq"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!trimmomatic SE -phred33 ENCFF000ASM.fastq ENCFF000ASM_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
        "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6jsjMTfGOpgk",
        "outputId": "6c09030d-9c56-450c-a954-dd4a53df22a3"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TrimmomaticSE: Started with arguments:\n",
            " -phred33 ENCFF000ASM.fastq ENCFF000ASM_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Automatically using 2 threads\n",
            "java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)\n",
            "\tat java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.io.FileInputStream.open(FileInputStream.java:195)\n",
            "\tat java.io.FileInputStream.<init>(FileInputStream.java:138)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
            "Input Reads: 48207516 Surviving: 40418814 (83.84%) Dropped: 7788702 (16.16%)\n",
            "TrimmomaticSE: Completed successfully\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!./FastQC/fastqc ENCFF000ASM_trimmed.fastq"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "SR7FKxvCOqLp",
        "outputId": "24fdd2da-a363-418c-ef20-84e889144a3a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF000ASM_trimmed.fastq\n",
            "Approx 5% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 10% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 15% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 20% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 25% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 30% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 35% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 40% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 45% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 50% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 55% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 60% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 65% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 70% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 75% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 80% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 85% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 90% complete for ENCFF000ASM_trimmed.fastq\n",
            "Approx 95% complete for ENCFF000ASM_trimmed.fastq\n",
            "Analysis complete for ENCFF000ASM_trimmed.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "W9QHWBLlpMjQ"
      },
      "source": [
        "## Выравнивание на хромосому\n",
        "\n",
        "Следует выбрать одну хромосому, потому что ресурсы google colab ограничены. В случае проблем, таких как Segmentation fault, error 139, следует выбрать хромосому меньшего размера.\n",
        "\n",
        "Последовательность нуклеотидов хромосомы можно скачать по адресу https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "N4JIGZOsGiY2",
        "outputId": "cce8c1b2-cd42-486d-9474-7486abc7ec2b"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-10 06:51:27--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr14.fa.gz\n",
            "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 29295890 (28M) [application/x-gzip]\n",
            "Saving to: ‘chr14.fa.gz’\n",
            "\n",
            "chr14.fa.gz         100%[===================>]  27.94M  17.4MB/s    in 1.6s    \n",
            "\n",
            "2022-03-10 06:51:29 (17.4 MB/s) - ‘chr14.fa.gz’ saved [29295890/29295890]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr14.fa.gz\n",
        "!gzip -d chr14.fa.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 23,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bhraQ6NdGiPN",
        "outputId": "d3cb20e2-2cc9-4222-eeca-fb69f3a8621c"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Settings:\n",
            "  Output files: \"chromosome_index.*.bt2\"\n",
            "  Line rate: 6 (line is 64 bytes)\n",
            "  Lines per side: 1 (side is 64 bytes)\n",
            "  Offset rate: 4 (one in 16)\n",
            "  FTable chars: 10\n",
            "  Strings: unpacked\n",
            "  Max bucket size: default\n",
            "  Max bucket size, sqrt multiplier: default\n",
            "  Max bucket size, len divisor: 4\n",
            "  Difference-cover sample period: 1024\n",
            "  Endianness: little\n",
            "  Actual local endianness: little\n",
            "  Sanity checking: disabled\n",
            "  Assertions: disabled\n",
            "  Random seed: 0\n",
            "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
            "Input files DNA, FASTA:\n",
            "  chr14.fa\n",
            "Building a SMALL index\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:01\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:01\n",
            "bmax according to bmaxDivN setting: 22642037\n",
            "Using parameters --bmax 16981528 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 16981528 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:02\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:01\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 9.05681e+07 (target: 16981527)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 1\n",
            "  No samples; assembling all-inclusive block\n",
            "  Sorting block of length 90568149 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:47\n",
            "Returning block of 90568150 for bucket 1\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 26673415\n",
            "fchr[G]: 45097173\n",
            "fchr[T]: 63656206\n",
            "fchr[$]: 90568149\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 34384187 bytes to primary EBWT file: chromosome_index.1.bt2\n",
            "Wrote 22642044 bytes to secondary EBWT file: chromosome_index.2.bt2\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 90568149\n",
            "    bwtLen: 90568150\n",
            "    sz: 22642038\n",
            "    bwtSz: 22642038\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 5660510\n",
            "    offsSz: 22642040\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 471710\n",
            "    numLines: 471710\n",
            "    ebwtTotLen: 30189440\n",
            "    ebwtTotSz: 30189440\n",
            "    color: 0\n",
            "    reverse: 0\n",
            "Total time for call to driver() for forward index: 00:01:07\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:00\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:01\n",
            "  Time to reverse reference sequence: 00:00:00\n",
            "bmax according to bmaxDivN setting: 22642037\n",
            "Using parameters --bmax 16981528 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 16981528 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:02\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:01\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 9.05681e+07 (target: 16981527)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 1\n",
            "  No samples; assembling all-inclusive block\n",
            "  Sorting block of length 90568149 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:45\n",
            "Returning block of 90568150 for bucket 1\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 26673415\n",
            "fchr[G]: 45097173\n",
            "fchr[T]: 63656206\n",
            "fchr[$]: 90568149\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 34384187 bytes to primary EBWT file: chromosome_index.rev.1.bt2\n",
            "Wrote 22642044 bytes to secondary EBWT file: chromosome_index.rev.2.bt2\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 90568149\n",
            "    bwtLen: 90568150\n",
            "    sz: 22642038\n",
            "    bwtSz: 22642038\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 5660510\n",
            "    offsSz: 22642040\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 471710\n",
            "    numLines: 471710\n",
            "    ebwtTotLen: 30189440\n",
            "    ebwtTotSz: 30189440\n",
            "    color: 0\n",
            "    reverse: 1\n",
            "Total time for backward call to driver() for mirror index: 00:01:04\n",
            "CPU times: user 1.11 s, sys: 155 ms, total: 1.26 s\n",
            "Wall time: 2min 11s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2-build chr14.fa chromosome_index"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 24,
      "metadata": {
        "id": "E_usI8XUrQnR"
      },
      "outputs": [],
      "source": [
        "!mkdir bowtie2_res"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "3t80UATHqFe7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "283e2350-1ead-4b71-e3ee-f40fe42f33a9"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "40418814 reads; of these:\n",
            "  40418814 (100.00%) were unpaired; of these:\n",
            "    33383834 (82.59%) aligned 0 times\n",
            "    1752589 (4.34%) aligned exactly 1 time\n",
            "    5282391 (13.07%) aligned >1 times\n",
            "17.41% overall alignment rate\n",
            "CPU times: user 9.56 s, sys: 997 ms, total: 10.6 s\n",
            "Wall time: 18min 21s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF000ASM_trimmed.fastq \\\n",
        "          -S bowtie2_res/res_ASM.sam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "so1jTdGHoEir",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4666a7c0-068a-4fc6-8e67-d54ca4dead5b"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "8383067 reads; of these:\n",
            "  8383067 (100.00%) were unpaired; of these:\n",
            "    6879801 (82.07%) aligned 0 times\n",
            "    354648 (4.23%) aligned exactly 1 time\n",
            "    1148618 (13.70%) aligned >1 times\n",
            "17.93% overall alignment rate\n",
            "CPU times: user 3.36 s, sys: 389 ms, total: 3.75 s\n",
            "Wall time: 6min 26s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF000ATK_trimmed.fastq \\\n",
        "          -S bowtie2_res/res_ATK.sam"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF000ARK_trimmed.fastq \\\n",
        "          -S bowtie2_res/res_ARK.sam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "HAOKvVALmRoP",
        "outputId": "3a5b7a41-488b-4e8d-9d0e-827468ed0746"
      },
      "execution_count": 25,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "6446684 reads; of these:\n",
            "  6446684 (100.00%) were unpaired; of these:\n",
            "    5221078 (80.99%) aligned 0 times\n",
            "    283250 (4.39%) aligned exactly 1 time\n",
            "    942356 (14.62%) aligned >1 times\n",
            "19.01% overall alignment rate\n",
            "CPU times: user 2.14 s, sys: 271 ms, total: 2.41 s\n",
            "Wall time: 4min 31s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ysoDrnSCrei0"
      },
      "source": [
        "Проанализируйте выдачу bowtie. Почему процент выравниваний получился именно таким?\n",
        "\n",
        "Имеет смысл для дальнейшего анализа отобрать уникально картированные риды."
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "\n",
        "df = pd.DataFrame({\n",
        "        'ChIP-seq': ['ENCFF000ASM_trimmed', 'ENCFF000ATK_trimmed','ENCFF000ARK_trimmed'],\n",
        "        'Количество ридов всего': [40418814, 8383067, 6446684],\n",
        "        'Количество ридов, которые выравнились уникально': [\"1752589 (4.34%)\", \"354648 (4.23%)\", \"283250 (4.39%)\"],\n",
        "        'Количество ридов, которые выравнились неуникально': [\"5282391 (13.07%)\", \"1148618 (13.70%)\", \"942356 (14.62%)\"],\n",
        "        'Количество ридов, которые не выравнились': [\"33383834 (82.59%)\", \"6879801 (82.07%)\", \"5221078 (80.99%)\"],\n",
        "     })\n",
        "df"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 203
        },
        "id": "Xy-_ImfsZovB",
        "outputId": "c2a5eaf3-ca66-4445-ec48-8205d23c445b"
      },
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "\n",
              "  <div id=\"df-d4e3db1e-10c6-4371-9dbb-16881e583ae7\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>ChIP-seq</th>\n",
              "      <th>Количество ридов всего</th>\n",
              "      <th>Количество ридов, которые выравнились уникально</th>\n",
              "      <th>Количество ридов, которые выравнились неуникально</th>\n",
              "      <th>Количество ридов, которые не выравнились</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>ENCFF000ASM_trimmed</td>\n",
              "      <td>40418814</td>\n",
              "      <td>1752589 (4.34%)</td>\n",
              "      <td>5282391 (13.07%)</td>\n",
              "      <td>33383834 (82.59%)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>ENCFF000ATK_trimmed</td>\n",
              "      <td>8383067</td>\n",
              "      <td>354648 (4.23%)</td>\n",
              "      <td>1148618 (13.70%)</td>\n",
              "      <td>6879801 (82.07%)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>ENCFF000ARK_trimmed</td>\n",
              "      <td>6446684</td>\n",
              "      <td>283250 (4.39%)</td>\n",
              "      <td>942356 (14.62%)</td>\n",
              "      <td>5221078 (80.99%)</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-d4e3db1e-10c6-4371-9dbb-16881e583ae7')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-d4e3db1e-10c6-4371-9dbb-16881e583ae7 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-d4e3db1e-10c6-4371-9dbb-16881e583ae7');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ],
            "text/plain": [
              "              ChIP-seq  Количество ридов всего  \\\n",
              "0  ENCFF000ASM_trimmed                40418814   \n",
              "1  ENCFF000ATK_trimmed                 8383067   \n",
              "2  ENCFF000ARK_trimmed                 6446684   \n",
              "\n",
              "  Количество ридов, которые выравнились уникально  \\\n",
              "0                                 1752589 (4.34%)   \n",
              "1                                  354648 (4.23%)   \n",
              "2                                  283250 (4.39%)   \n",
              "\n",
              "  Количество ридов, которые выравнились неуникально  \\\n",
              "0                                  5282391 (13.07%)   \n",
              "1                                  1148618 (13.70%)   \n",
              "2                                   942356 (14.62%)   \n",
              "\n",
              "  Количество ридов, которые не выравнились  \n",
              "0                        33383834 (82.59%)  \n",
              "1                         6879801 (82.07%)  \n",
              "2                         5221078 (80.99%)  "
            ]
          },
          "metadata": {},
          "execution_count": 26
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "TmfJKT0rmLyp"
      },
      "source": [
        "Преобразование .sam в .bam (не требуется если запускать macs2 на .sam файлах)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "2PteJ-KVqFhs"
      },
      "outputs": [],
      "source": [
        "!samtools view -h -S -b -o bowtie2_res/res_ASM.bam bowtie2_res/res_ASM.sam\n",
        "!samtools view -h -S -b -o bowtie2_res/res_ATK.bam bowtie2_res/res_ATK.sam"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YZOlAXrpdN1t",
        "outputId": "62772812-201a-499e-ed75-bf0315a699b1"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mounted at /content/drive\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "eLtG7xtRup19"
      },
      "source": [
        "На этом моменте можно скачать себе на компьютер все полученные .bam или .sam файлы на случай сбоя."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XZtQrPZMmgnC"
      },
      "source": [
        "## Peak calling\n",
        "\n",
        "Помимо macs2 может быть использован пакет Homer для поиска пиков, а также macs3."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "uX1ps4VDp_Ae"
      },
      "outputs": [],
      "source": [
        "!mkdir macs2"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "EJQOCcv2nJW_",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "9f158385-6db7-4e2e-9fa7-79bab0d2d341"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Tue, 01 Mar 2022 21:52:11: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_ASM.sam -c bowtie2_res/res_ATK.sam -f SAM --outdir macs2\n",
            "# ARGUMENTS LIST:\n",
            "# name = NA\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_ASM.sam']\n",
            "# control file = ['bowtie2_res/res_ATK.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Tue, 01 Mar 2022 21:52:11: #1 read tag files... \n",
            "INFO  @ Tue, 01 Mar 2022 21:52:11: #1 read treatment tags... \n",
            "INFO  @ Tue, 01 Mar 2022 21:52:15:  3000000 \n",
            "INFO  @ Tue, 01 Mar 2022 21:52:20:  8000000 \n",
            "INFO  @ Tue, 01 Mar 2022 21:52:21:  9000000 \n",
            "INFO  @ Tue, 01 Mar 2022 21:52:49:  32000000 \n",
            "INFO  @ Tue, 01 Mar 2022 21:52:57:  34000000 \n",
            "INFO  @ Tue, 01 Mar 2022 21:52:58:  35000000 \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:05: #1.2 read input tags... \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #1 tag size is determined as 36 bps \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #1 tag size = 36.0 \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #1  total tags in treatment: 7034980 \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #1 user defined the maximum tags... \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #1  tags after filtering in treatment: 5152535 \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #1  Redundant rate of treatment: 0.27 \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #1  total tags in control: 1503266 \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #1 user defined the maximum tags... \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #1  tags after filtering in control: 1384707 \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #1  Redundant rate of control: 0.08 \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #1 finished! \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #2 Build Peak Model... \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:15: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:16: #2 number of paired peaks: 32154 \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:16: start model_add_line... \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:17: start X-correlation... \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:17: end of X-cor \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:17: #2 finished! \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:17: #2 predicted fragment length is 36 bps \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:17: #2 alternative fragment length(s) may be 36 bps \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:17: #2.2 Generate R script for model : macs2/NA_model.r \n",
            "WARNING @ Tue, 01 Mar 2022 21:53:17: #2 Since the d (36) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Tue, 01 Mar 2022 21:53:17: #2 You may need to consider one of the other alternative d(s): 36 \n",
            "WARNING @ Tue, 01 Mar 2022 21:53:17: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:17: #3 Call peaks... \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:17: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:17: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:27: #3 Call peaks for each chromosome... \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:32: #4 Write output xls file... macs2/NA_peaks.xls \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:32: #4 Write broad peak in broadPeak format file... macs2/NA_peaks.broadPeak \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:32: #4 Write broad peak in bed12/gappedPeak format file... macs2/NA_peaks.gappedPeak \n",
            "INFO  @ Tue, 01 Mar 2022 21:53:32: Done! \n"
          ]
        }
      ],
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_ASM.sam \\\n",
        "    -c bowtie2_res/res_ATK.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "HrNwxs7ArPz7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4742f827-2224-40c2-9416-5144348c911e"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-01 22:04:48--  https://www.encodeproject.org/files/ENCFF453PEP/@@download/ENCFF453PEP.bed.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2017/01/11/2d0a7961-8edc-47f0-a32a-d9e1f4e5455f/ENCFF453PEP.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF453PEP.bed.gz&AWSAccessKeyId=ASIATGZNGCNX6DYX23QR&Signature=4lEK8957V7wsopG8xgQvaVXQMLQ%3D&x-amz-security-token=IQoJb3JpZ2luX2VjELb%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQC4EYvFRZuHX7WBJFpa6gv%2BduujYCtsThM5WiQcLkDyAAIgIn2xwVlAtmOzjaWmT7PXbyBGH6EMkMEMdo5wcVFI6csq%2BgMIHxAAGgwyMjA3NDg3MTQ4NjMiDFXBKzicxAhN%2FyhbiSrXA6V%2FSjOAmFZ0nwqKpOiOB28o94kybwRlj8U89boOXOfyZwHQ586aGQnX1gTzes4BZaBfcQL6s6Eqhr3uejG5IqZWSe33gj5Bxi0NjcwJglkVn%2FN7cKWiTZuk8AJ2krTkmrXWXXSLFLaRSfgg2STBvAhhmZAj%2BTTctP3aSRRpjxIilbOLRU5TkGj9GtrcZPZCQ5cwqF9QlKPdXVYJ6Jz6bj57995LJIJZf9K%2BfAlmNxJEnU2VUYG%2Bp9GlkPtD8Kqzb1Dasm6tegTVd5T1BeVC6HMPkDDhc63%2B5Lbm1FBsOToCFYGNOUwGrTeM4nqNx25zWCrl144kYphOwSpJrnkL18cCPt5De0ENR%2FPR76wJeWZgBljh7Oqh5qfswB%2BdAkDCf49LwQs%2Fm6n0On4%2B4Uel%2FIkL8KNsQIKNSX0ONjWf6zv3Tasgf6XdZ1rOv4H5CScvoJJIJPEVbdh37yGeHk2r3bJQDaUs6cxkoVqmv5g%2BDP%2FgMqNMFsNJ%2BlWb3i3TbFanKM6LEByRBm02kBlFCUH1lICQmpSqVr5LRmtOI44hqSbVv1SknlGaPIDgH%2ByUWwWvuKM9vAq%2B0HrXA6qIYnf%2BnFGkOzddu1KUJG3XGmYMEGj0jyUwPwEGiDDxnPqQBjqlASLmpmJONw9uERW5r35bOCKmLRBOJHkmRKBvn1jiXFIiWGmzzO%2B8KzHVFKhHZr6%2Bt%2BP2EBV7wfAWPv9RYaWZH3fWu%2B4cYmWfJoICMuZfaLVRMQ9Oq3c8rZigNjwdbu7Ar5GxmXWyTjf%2FaEwuR0Y4NABNPHiaLFc1fp44ADcXSaEnMyoA84yHaakIsIssNcD9jLLva2Bd3ouJngpSgrdp1eUY1Zt6XQ%3D%3D&Expires=1646301888 [following]\n",
            "--2022-03-01 22:04:48--  https://encode-public.s3.amazonaws.com/2017/01/11/2d0a7961-8edc-47f0-a32a-d9e1f4e5455f/ENCFF453PEP.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF453PEP.bed.gz&AWSAccessKeyId=ASIATGZNGCNX6DYX23QR&Signature=4lEK8957V7wsopG8xgQvaVXQMLQ%3D&x-amz-security-token=IQoJb3JpZ2luX2VjELb%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQC4EYvFRZuHX7WBJFpa6gv%2BduujYCtsThM5WiQcLkDyAAIgIn2xwVlAtmOzjaWmT7PXbyBGH6EMkMEMdo5wcVFI6csq%2BgMIHxAAGgwyMjA3NDg3MTQ4NjMiDFXBKzicxAhN%2FyhbiSrXA6V%2FSjOAmFZ0nwqKpOiOB28o94kybwRlj8U89boOXOfyZwHQ586aGQnX1gTzes4BZaBfcQL6s6Eqhr3uejG5IqZWSe33gj5Bxi0NjcwJglkVn%2FN7cKWiTZuk8AJ2krTkmrXWXXSLFLaRSfgg2STBvAhhmZAj%2BTTctP3aSRRpjxIilbOLRU5TkGj9GtrcZPZCQ5cwqF9QlKPdXVYJ6Jz6bj57995LJIJZf9K%2BfAlmNxJEnU2VUYG%2Bp9GlkPtD8Kqzb1Dasm6tegTVd5T1BeVC6HMPkDDhc63%2B5Lbm1FBsOToCFYGNOUwGrTeM4nqNx25zWCrl144kYphOwSpJrnkL18cCPt5De0ENR%2FPR76wJeWZgBljh7Oqh5qfswB%2BdAkDCf49LwQs%2Fm6n0On4%2B4Uel%2FIkL8KNsQIKNSX0ONjWf6zv3Tasgf6XdZ1rOv4H5CScvoJJIJPEVbdh37yGeHk2r3bJQDaUs6cxkoVqmv5g%2BDP%2FgMqNMFsNJ%2BlWb3i3TbFanKM6LEByRBm02kBlFCUH1lICQmpSqVr5LRmtOI44hqSbVv1SknlGaPIDgH%2ByUWwWvuKM9vAq%2B0HrXA6qIYnf%2BnFGkOzddu1KUJG3XGmYMEGj0jyUwPwEGiDDxnPqQBjqlASLmpmJONw9uERW5r35bOCKmLRBOJHkmRKBvn1jiXFIiWGmzzO%2B8KzHVFKhHZr6%2Bt%2BP2EBV7wfAWPv9RYaWZH3fWu%2B4cYmWfJoICMuZfaLVRMQ9Oq3c8rZigNjwdbu7Ar5GxmXWyTjf%2FaEwuR0Y4NABNPHiaLFc1fp44ADcXSaEnMyoA84yHaakIsIssNcD9jLLva2Bd3ouJngpSgrdp1eUY1Zt6XQ%3D%3D&Expires=1646301888\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.130.129\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.130.129|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1890027 (1.8M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF453PEP.bed.gz’\n",
            "\n",
            "ENCFF453PEP.bed.gz  100%[===================>]   1.80M  3.68MB/s    in 0.5s    \n",
            "\n",
            "2022-03-01 22:04:49 (3.68 MB/s) - ‘ENCFF453PEP.bed.gz’ saved [1890027/1890027]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF453PEP/@@download/ENCFF453PEP.bed.gz\n",
        "!gzip -d ENCFF453PEP.bed.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "MYEGEbQjvEwY"
      },
      "source": [
        "## Сравнение результатов\n",
        "\n",
        "Сравниваем те пики, которые мы получили, с пиками, которые приведены в ENCODE (важно, чтобы версии генома hg38 или hg19) для .bed файла из ENCODE и той хромосомы, которую скачивали выше, сопадали.\n",
        "\n",
        "Проанализируйте полученные результаты и приведите свои рассуждения в README.md. Как можно объяснить различия в количестве пересечений?"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "_1EAygSCvc8a",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "3eae5b9e-5871-4dc2-ae44-c73927b1fac3"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn1.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i macs2/NA_peaks.broadPeak ENCFF453PEP.bed --filenames --output venn_results/venn1.jpg"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7qqn3n2cRdkt",
        "outputId": "5dcf38ae-81db-4407-d3bd-0c1babd5f3e5"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn2.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i ENCFF453PEP.bed macs2/NA_peaks.broadPeak --filenames --output venn_results/venn2.jpg"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        ""
      ],
      "metadata": {
        "id": "isYx230PAANq"
      },
      "execution_count": null,
      "outputs": []
    }
  ],
  "metadata": {
    "colab": {
      "collapsed_sections": [],
      "name": "ChIP_Ш.ipynb",
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}